Created a new tool for creating Incidents with email

samikroy opened the pull request:
A detailed explanation is here.
https://github.com/samikroy/Azure-Sentinel/blob/patch-32/Tools/Create%20Incidents%20with%20Email/Readme.md

This will be a part of the readme after the PR merge.

samikroy commented:
@v-mchatla  - Request your help for the approval.

v-spadarthi commented:
@rushriva : Please have a look and provide an update. Thanks!

v-spadarthi commented:
@rushriva : Please have a look and provide an update. Thanks!

v-spadarthi commented:
@rushriva : Please have a look and provide an update. Thanks!

v-spadarthi commented:
@rushriva is checking with @shainw.

v-spadarthi commented:
@rushriva : Please have a look and provide an update. Thanks!

v-spadarthi commented:
@rushriva : Please have a look and provide an update. Thanks!

v-spadarthi commented:
@rushriva : Please have a look and provide an update. Thanks!

v-spadarthi commented:
@rushriva : Please have a look and provide an update. Thanks!

v-spadarthi commented:
@rushriva : Please have a look and provide an update. Thanks!

v-spadarthi commented:
@rushriva : Please have a look and provide an update. Thanks!

v-spadarthi commented:
@rushriva : Please have a look and provide an update. Thanks!

v-spadarthi commented:
@rushriva : Please have a look and provide an update. Thanks!

v-spadarthi commented:
@rushriva : Please have a look and provide an update. Thanks!

anki-narravula commented:
Hi @samikroy  - 

Thank you for this content. Based on the functionality I see this is automation playbook and can we move this to Playbook folder, so that we can get into [Content hub](https://techcommunity.microsoft.com/t5/microsoft-sentinel-blog/introducing-microsoft-sentinel-content-hub/ba-p/2928102) subsequently.  While moving, can you please add following elements, so that it will be ready for content hub

1. Metadata section: 

- You can find the instructions here - https://github.com/Azure/Azure-Sentinel/tree/master/docs/New%20Playbooks%20Contribution%20Guide#contribution-guidelines
- For examples you can refer this playbook https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Okta%20Single%20Sign-On/Playbooks/OktaPlaybooks/Okta-EnrichIncidentWithUserDetails/azuredeploy.json 
2. As a best practice, we recommend sentinel connection in playbooks uses "ManagedSecurityIdentity".
Please refer Sample [Template](https://github.com/Azure/Azure-Sentinel/blob/master/Playbooks/.template/incident-trigger/azuredeploy.json) for sample template. For more details, refer [this](https://techcommunity.microsoft.com/t5/azure-sentinel/what-s-new-managed-identity-for-azure-sentinel-logic-apps/ba-p/2068204). Make sure to do changes at 3 places 

![image](https://user-images.githubusercontent.com/10404181/141414755-5b90164b-ae6c-415d-9ee2-4a2725ea6cd4.png)

![image](https://user-images.githubusercontent.com/10404181/141414637-b4c76777-4201-43e7-929f-4389cca0ef19.png)

![image](https://user-images.githubusercontent.com/10404181/141414457-2bb8c70a-7e3c-4897-8acb-84eeaba62681.png)

v-spadarthi commented:
@samikroy : Please address the @anki-narravula comments.

v-spadarthi commented:
@anki-narravula : Please have a look and provide your feedback @samikroy addressed your comments.

anki-narravula commented:
> @anki-narravula @v-spadarthi - Please have a look at the updated code and share your reviews !

@samikroy  Metadata object is still missing, example one

![image](https://user-images.githubusercontent.com/10404181/205893495-109d1aca-e1a2-4201-89dc-07a00dd59eec.png)

v-spadarthi commented:
@samikroy : Please address the @anki-narravula comments.

v-spadarthi commented:
@samikroy : Please address the @anki-narravula comments.

v-mchatla commented:
hi @samikroy,
Please add metadata object to your playbook, so that @anki-narravula can review and merge your changes.
Thanks

samikroy commented:
@v-mchatla - Could you please help with the merge.

samikroy commented:
@anki-narravula , @v-spadarthi & @v-mchatla - Could you please help with the merge as the comments are addressed now.

v-mchatla commented:
Hi @anki-narravula,
Author has addressed your comments. Can you please review and provide your approval.
Thanks

samikroy commented:
@v-mchatla , @anki-narravula, @v-spadarthi - Could you please help with the merge.

manishkumar1991 commented:
> Hi @samikroy -
> 
> Thank you for this content. Based on the functionality I see this is automation playbook and can we move this to [Playbook folder](https://github.com/Azure/Azure-Sentinel/tree/master/Playbooks), so that we can get into [Content hub](https://techcommunity.microsoft.com/t5/microsoft-sentinel-blog/introducing-microsoft-sentinel-content-hub/ba-p/2928102) subsequently. While moving, can you please add following elements, so that it will be ready for content hub
> 
> 1. Metadata section:
> 
> * You can find the instructions here - https://github.com/Azure/Azure-Sentinel/tree/master/docs/New%20Playbooks%20Contribution%20Guide#contribution-guidelines
> * For examples you can refer this playbook https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Okta%20Single%20Sign-On/Playbooks/OktaPlaybooks/Okta-EnrichIncidentWithUserDetails/azuredeploy.json
> 
> 2. As a best practice, we recommend sentinel connection in playbooks uses "ManagedSecurityIdentity".
>    Please refer Sample [Template](https://github.com/Azure/Azure-Sentinel/blob/master/Playbooks/.template/incident-trigger/azuredeploy.json) for sample template. For more details, refer [this](https://techcommunity.microsoft.com/t5/azure-sentinel/what-s-new-managed-identity-for-azure-sentinel-logic-apps/ba-p/2068204). Make sure to do changes at 3 places
> 
> ![image](https://user-images.githubusercontent.com/10404181/141414755-5b90164b-ae6c-415d-9ee2-4a2725ea6cd4.png)
> 
> ![image](https://user-images.githubusercontent.com/10404181/141414637-b4c76777-4201-43e7-929f-4389cca0ef19.png)
> 
> ![image](https://user-images.githubusercontent.com/10404181/141414457-2bb8c70a-7e3c-4897-8acb-84eeaba62681.png)

@samikroy , did you checked the 2-point mentioned by anki , regarding "managedserviceidentity" , As i am not able to see the changes being done, 

Kindly check again

manishkumar1991 commented on Playbooks/Create Incidents with Email/Readme.md:
```diff
@@ -0,0 +1,46 @@
+
+### Deployment
+
+Once deployed you can configure the connetions as below
```
Please check spellings: connections

samikroy replied:
addressed,

manishkumar1991 commented on Playbooks/Create Incidents with Email/Readme.md:
```diff
@@ -0,0 +1,46 @@
+
+
```
Readme file will look good, if you can mention and explain, the use cases where this playbook can be utilized in infra or SOC , SOAR , if possible

samikroy replied:
added.

manishkumar1991 commented on Playbooks/Create Incidents with Email/Readme.md:
```diff
@@ -0,0 +1,46 @@
+### Configuration
+
+1. Sentinel Connection
+
```
Screenshot is good , but it will look nicer if you can provide the step by step approach to create and configured the connection

samikroy replied:
Generally connections are configured during the ARM template deployment.
This is a fallback step added to ensure the tool does not break.
Furthermore, if this logic app is expected to be configured as solution, this separate configuration will not be necessary.
However, in case you suggest this as a mandatory step for the PR merge, please provide an example and will be able to update the PR.
Thank you & apologies for being late in the response.

manishkumar1991 commented on Playbooks/Create Incidents with Email/Readme.md:
```diff
@@ -0,0 +1,46 @@
+
+
+2. Office 365 Connection
+
```
same here 

Screenshot is good , but it will look nicer if you can provide the step by step approach to create and configured the connection

samikroy replied:
Same as above.

v-spadarthi commented:
@samikroy: Please address the @manishkumar1991 comments. Thanks!

v-spadarthi commented:
@samikroy: Please address the @manishkumar1991 comments. Thanks!

v-spadarthi commented:
@samikroy: Please address the @manishkumar1991 comments. Thanks!

v-spadarthi commented:
@samikroy: Please address the @manishkumar1991 comments. Thanks!

v-atulyadav commented:
Hi @samikroy, we still waiting your response for @manishkumar1991 comments. Thanks

v-atulyadav commented:
Hi @samikroy, I would appreciate it if you shared the details with @manishkumar1991 that he asks for. Thanks.

v-atulyadav commented:
Hi @samikroy, we still waiting your response for @manishkumar1991 comments. Thanks

v-atulyadav commented:
Hi @samikroy, please address the @manishkumar1991 comments. Thanks!

v-atulyadav commented:
Hi @samikroy, we wanted to check on the status of PR https://github.com/Azure/Azure-Sentinel/pull/6409. PR is pending for more than 20 days. Please let us know if you need any assistance to review this PR. Per our standard operating procedures if no response is received in the next 7 business days, we will close this PR. Thank you for your cooperation.

v-atulyadav commented:
Hi @samikroy, we wanted to check on the status of PR https://github.com/Azure/Azure-Sentinel/pull/6409. PR is pending for more than 22 days. Please let us know if you need any assistance to review this PR. Per our standard operating procedures if no response is received in the next 7 business days, we will close this PR. Thank you for your cooperation

v-atulyadav commented:
Hi @samikroy, we wanted to check on the status of PR https://github.com/Azure/Azure-Sentinel/pull/6409. PR is pending for more than 24 days. Please let us know if you need any assistance to review this PR. Per our standard operating procedures if no response is received in the next 7 business days, we will close this PR. Thank you for your cooperation

samikroy commented:
@v-atulyadav  & @manishkumar1991 - Please have a look the updated comment and revert for any queries.

v-atulyadav commented:
Hi @manishkumar1991, @samikroy has responded on your comments please check and respond accordingly. Thanks.

v-atulyadav commented:
Hi @manishkumar1991, @samikroy has responded on your comments please check and respond accordingly. Thanks.

v-atulyadav commented:
Hi @manishkumar1991, check @samikroy's response to your comments. Thanks.

v-atulyadav commented:
Hi @manishkumar1991, @samikroy has responded on your comments please check and respond accordingly. Thanks.

v-atulyadav commented:
Hi @manishkumar1991, please provide your feedback. Thanks.

v-atulyadav commented:
Hi @manishkumar1991, @samikroy has responded on your comments please check and respond accordingly. Thanks.

manishkumar1991 commented:
> Hi @manishkumar1991, @samikroy has responded on your comments please check and respond accordingly. Thanks.

checking

manishkumar1991 commented:
@samikroy ,

Please address comments given by @anki-narravula regarding the use of "managed service identity " while making azure sentinel connection in playbook .

@v-atulyadav : Readme file changes has been taken care with minor correction needed, but first comments mentioned by anki is still not addressed .

manishkumar1991 commented on Playbooks/Create Incidents with Email/Readme.md:
```diff
@@ -0,0 +1,52 @@
+
+### Use cases
+
+SOC - In a organization there might be Security Incident / Suspecious activity occurring to the resources where Security is not tightened up yet. 
```
check   Suspecious  Spell mistake.

manishkumar1991 commented:
> Hi @samikroy -
> 
> Thank you for this content. Based on the functionality I see this is automation playbook and can we move this to [Playbook folder](https://github.com/Azure/Azure-Sentinel/tree/master/Playbooks), so that we can get into [Content hub](https://techcommunity.microsoft.com/t5/microsoft-sentinel-blog/introducing-microsoft-sentinel-content-hub/ba-p/2928102) subsequently. While moving, can you please add following elements, so that it will be ready for content hub
> 
> 1. Metadata section:
> 
> * You can find the instructions here - https://github.com/Azure/Azure-Sentinel/tree/master/docs/New%20Playbooks%20Contribution%20Guide#contribution-guidelines
> * For examples you can refer this playbook https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Okta%20Single%20Sign-On/Playbooks/OktaPlaybooks/Okta-EnrichIncidentWithUserDetails/azuredeploy.json
> 
> 2. As a best practice, we recommend sentinel connection in playbooks uses "ManagedSecurityIdentity".
>    Please refer Sample [Template](https://github.com/Azure/Azure-Sentinel/blob/master/Playbooks/.template/incident-trigger/azuredeploy.json) for sample template. For more details, refer [this](https://techcommunity.microsoft.com/t5/azure-sentinel/what-s-new-managed-identity-for-azure-sentinel-logic-apps/ba-p/2068204). Make sure to do changes at 3 places
> 
> ![image](https://user-images.githubusercontent.com/10404181/141414755-5b90164b-ae6c-415d-9ee2-4a2725ea6cd4.png)
> 
> ![image](https://user-images.githubusercontent.com/10404181/141414637-b4c76777-4201-43e7-929f-4389cca0ef19.png)
> 
> ![image](https://user-images.githubusercontent.com/10404181/141414457-2bb8c70a-7e3c-4897-8acb-84eeaba62681.png)

@samikroy These comments still not addressed, can you please make the changes

v-atulyadav commented:
Hi @samikroy, please addressed comments above. Thanks

v-atulyadav commented:
Hi @samikroy, could you please take a look at the above comments. Thanks

v-atulyadav commented:
Hi @samikroy, I would appreciate if you could take a look at the above comments Thanks.

samikroy commented:
@v-atulyadav & @manishkumar1991 - Have fixed the mentioned ones.
As there a lot of back and forth could you pease mention the exact error with line numbers mentioned to fast track this 
Thank you.

manishkumar1991 commented:
> @v-atulyadav & @manishkumar1991 - Have fixed the mentioned ones. As there a lot of back and forth could you please mention the exact error with line numbers mentioned to fast track this Thank you.

@samikroy :As PR approver, We can only suggest you good practice, so that good quality content gets merged in our branch.

I see that if you can scroll up and see that Anki from our team, has already told you on the first day of the PR itself, and mentioned what needs to be changed. If proper action has been taken at that time, we could not have gone back and forth too much.

2. Kindly change the AzureSentinelConnection with MicrosoftSentinelConnection at all places in your arm template.

Thanks
Manish Kumar

manishkumar1991 commented on Playbooks/Create Incidents with Email/azuredeploy.json:
```diff
@@ -0,0 +1,160 @@
+            "properties": {
+                "displayName": "[variables('AzureSentinelConnection')]",
+                "customParameterValues": {},
+                "api": {
```
@samikroy is this done? Please confirm

manishkumar1991 replied:
@samikroy 

Hello ,

after line number 58 you need to make the changes, you didn't get lost, that why I marked the comment at line 58. 

Changed mentioned above are needs to be added in Microsoft.Web/connections of sentinel connection not in office 365 connection. 

![image](https://user-images.githubusercontent.com/97503740/224624952-e793109a-ae85-46a9-890c-7d2f477ac983.png)

manishkumar1991 replied:
@v-atulyadav  Waiting for @samikroy to complete the changes properly

v-atulyadav commented:
Hi @samikroy, please respond on remaining fixes suggested by Anki. Thanks

samikroy commented:
> Hi @samikroy, please respond on remaining fixes suggested by Anki. Thanks

@v-atulyadav - Have address  the comments. Please let me know for any further changes .

v-atulyadav commented:
Hi @manishkumar1991, @samikroy has responded to comments please check. Thanks

manishkumar1991 commented:
@samikroy  , , Changes are not done, as asked .
Kindly refer to me previous comments

manishkumar1991 reviewed (approved):
approved

samikroy commented:
@manishkumar1991 , @v-atulyadav , @v-dvedak  - Thank you for the your time throughout the PR timeline with the detailed reviews till the approval & merge.
Apologies for the back and forth while addressing the comments.
